### DISCUSSION TITLE
Dynamic content for modal without reloading whole page

### DISCUSSION THREAD

**Discussion by sathia-musso** [livewire/livewire#5372] [1. Help] [answered]
I have a pretty big sized DOM structure that takes about 1 second to load (it’s more than 500 images) I want to open a detail modal onclick for each of these images without have to reload the whole DOM again. Is there a way to just re-render the modal part (ie: giving it, show:true and the array of data to display) I see from the network calls that the close method for the modal is just 1.5kb so I assume there must be a way to do it.

p.s.
I’m also using computed properties such as
`return view(‘livewire.feeds’)->with([‘videos’ => $this->videos]);
`
thanks for any help

**Comment by joshhanley** [comment, marked as answer]
@sathio you could make use of nested components, as once they are running in the page, they are isolated from each other. So you could have a modal component that handles showing the details of the image, which won't trigger a re-render of the main component.

Also you shouldn't have to pass computed properties into the view, you can access them directly in blade as `$this->videos` instead of `$videos`. See here for details https://laravel-livewire.com/docs/2.x/properties#computed-properties

Hope this helps!

**Reply by sathia-musso** [reply]
thank you, I will check tomorrow with a clear mind.

**Reply by joshhanley** [reply]
No worries! 🙂

**Reply by sathia-musso** [reply]
hi, were you suggesting to create 500 modal components in the dom? this would make it very busy, I don't like this approach too much. I was naively thinking that I could do something like: 

    @livewire('modal')

and then something like 

    wire:click="modal.openModal({{$video['id']}})" 

or something along the lines of emitTo, but I can't get it to work, maybe it's not supported at all?

thanks again for your help

**Reply by joshhanley** [reply]
@sathio no, a single modal component would be fine 🙂

On your photos you would emit an event to the modal component, like `wire:click="$emitTo('modal', 'open', '{{ $video['id'] }}')"`

Then listen for that in your modal component to open up, get the ID and show the details.

Hope this helps!

**Reply by sathia-musso** [reply]
thank you so much, in my tests I actually tried emitTo, but I didn't realise I had to make use of the listeners on the component. now it works perfectly. cheers

**Reply by joshhanley** [reply]
@sathio no worries! 🙂
